### PR TITLE
Add m3ninx/AllQuery

### PR DESCRIPTION
**Pull request by prateek**

### CODE REVIEW

File: src/m3ninx/search/query/all.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import (
+	"fmt"
+
+	"github.com/m3db/m3/src/m3ninx/generated/proto/querypb"
+	"github.com/m3db/m3/src/m3ninx/search"
+	"github.com/m3db/m3/src/m3ninx/search/searcher"
+)
+
+// AllQuery returns a query which matches all the known documents.
```
Comment (arnikola) [inline src/m3ninx/search/query/all.go]: nit: `all known documents`

File: src/m3ninx/search/query/codec.go [hunk 2]
```diff
@@ -85,8 +88,7 @@ func unmarshal(q *querypb.Query) (search.Query, error) {
 			qs = append(qs, sqry)
 		}
 		return NewDisjunctionQuery(qs), nil
-
 	}
 
-	return nil, fmt.Errorf("unknown query: %v", q)
+	return nil, fmt.Errorf("unknown query: %+v", q)
```
Comment (arnikola) [inline src/m3ninx/search/query/codec.go]: Wow, didn't know about this, it's pretty neat :+1:

**Comment by arnikola** [comment]
> Whats the point of this P.R? Is it just so we have an easy way of special-casing these types of queries?

I think it's primarily to make an explicit query type for tag matching v.s. regex matching `.*` or something; in future, could be useful for specifying the special case of listing all tags without providing a query

File: src/m3ninx/search/query/codec.go (hunk 2 above)
Comment (prateek) [reply src/m3ninx/search/query/codec.go]: lol :)

File: src/m3ninx/search/query/all.go (hunk 1 above)
Comment (prateek) [reply src/m3ninx/search/query/all.go]: cheers

**Comment by prateek** [comment]
Pretty much what you guys are saying - need an explicit way for users of `m3db/client` to be able to specify the lack of any filtering for aggregation queries

**Pull request merged**